src/Compilers/CSharp/Test/Semantic/Semantics/SimpleProgramsTests.cs

```diff
@@ -6233,5 +6237,333 @@ public void MissingTypes_02()
+                Diagnostic(ErrorCode.ERR_PredefinedTypeNotFound).WithArguments("System.Threading.Tasks.Task`1").WithLocation(1, 1),
+                // error CS5001: Program does not contain a static 'Main' method suitable for an entry point
+                Diagnostic(ErrorCode.ERR_NoEntryPoint).WithLocation(1, 1),
+                // (2,1): warning CS0028: '<simple-program-entry-point>' has the wrong signature to be an entry point
+                // await System.Threading.Tasks.Task.Factory.StartNew(() => "5");
+                Diagnostic(ErrorCode.WRN_InvalidMainSig, @"await System.Threading.Tasks.Task.Factory.StartNew(() => ""5"");
+return 11;
+").WithArguments("<simple-program-entry-point>").WithLocation(2, 1)
+                );
+        }
+
+        [Fact]
+        public void MissingTypes_06()
+        {
+            var text = @"
+await System.Threading.Tasks.Task.Factory.StartNew(() => ""5"");
+return 11;
+";
+
+            var comp = CreateCompilation(text, options: TestOptions.DebugExe, parseOptions: DefaultParseOptions);
+            comp.MakeTypeMissing(SpecialType.System_Int32);
+            comp.MakeTypeMissing(WellKnownType.System_Threading_Tasks_Task_T);
+            var entryPoint = SimpleProgramNamedTypeSymbol.GetSimpleProgramEntryPoint(comp);
+            Assert.Equal("System.Threading.Tasks.Task<System.Int32[missing]>[missing]", entryPoint.ReturnType.ToTestDisplayString());
+            Assert.False(entryPoint.ReturnsVoid);
+            comp.VerifyEmitDiagnostics(
+                // error CS0518: Predefined type 'System.Threading.Tasks.Task`1' is not defined or imported
+                Diagnostic(ErrorCode.ERR_PredefinedTypeNotFound).WithArguments("System.Threading.Tasks.Task`1").WithLocation(1, 1),
+                // error CS0518: Predefined type 'System.Int32' is not defined or imported
+                Diagnostic(ErrorCode.ERR_PredefinedTypeNotFound).WithArguments("System.Int32").WithLocation(1, 1),
+                // error CS5001: Program does not contain a static 'Main' method suitable for an entry point
+                Diagnostic(ErrorCode.ERR_NoEntryPoint).WithLocation(1, 1),
+                // (2,1): warning CS0028: '<simple-program-entry-point>' has the wrong signature to be an entry point
+                // await System.Threading.Tasks.Task.Factory.StartNew(() => "5");
+                Diagnostic(ErrorCode.WRN_InvalidMainSig, @"await System.Threading.Tasks.Task.Factory.StartNew(() => ""5"");
+return 11;
+").WithArguments("<simple-program-entry-point>").WithLocation(2, 1),
+                // (3,8): error CS0518: Predefined type 'System.Int32' is not defined or imported
+                // return 11;
+                Diagnostic(ErrorCode.ERR_PredefinedTypeNotFound, "11").WithArguments("System.Int32").WithLocation(3, 8)
+                );
+        }
+
+        [Fact]
+        public void Return_01()
+        {
+            var text = @"
+System.Console.WriteLine(""Hi!"");
+return;
+";
```

Comment:
Are we testing unreachable code, after `return;` for instance?

Reply:
> Are we testing unreachable code, after return; for instance?

Not specifically. There is a test that verifies that we run flow analysis for top level statements. Or do you have something else in mind?

---
In reply to: [410414111](https://github.com/dotnet/roslyn/pull/43451#discussion_r410414111) [](ancestors = 410414111)

Reply:
Added tests

---
In reply to: [410417194](https://github.com/dotnet/roslyn/pull/43451#discussion_r410417194) [](ancestors = 410417194,410414111)